lib/plugin-loader.js

```diff
@@ -752,71 +756,275 @@ PluginLoader.prototype = {
+          const neededEndpoints = acceptedComponents[AGENT_COMP_ID].endpoints;
+
+          if (acceptedOS && acceptedOS.length > 0 && components[AGENT_COMP_ID].os) { // Plugin has manually defined their OS support section
+            if (acceptedOS.includes(components[AGENT_COMP_ID].os)) {
+              console.log(`(${components[AGENT_COMP_ID].os}) is a supported platform for the agent to install (${plugin.identifier}).`);
+            } 
+            else if (acceptedOS.includes("!"+components[AGENT_COMP_ID].os)) {
+              plugin.error = new Error(`(${components[AGENT_COMP_ID].os}) is not a supported platform for the agent. Skipping (${plugin.identifier})... Supported: ${acceptedOS}`);
+            }
+            else {
+              console.log(`Agent could not verify (${components[AGENT_COMP_ID].os}) as a supported platform to install (${plugin.identifier}). Proceeding anyway...`);
+            }
+          }
+          if (setVersions && setVersions.length > 0 && components[AGENT_COMP_ID].version) {
+            const actualVer = components[AGENT_COMP_ID].version.split('.');
+            let supported = false;
+            // TODO: Implement complex version verification
+            // for (let i = 0; i < setVersions.length; i++) {
+            //   let compareVer = setVersions[i].split('.');
+
+            //   if (("!" + actualVer[0]).localeCompare(compareVer[0])) {
+            //     if (actualVer[1].localeCompare(compareVer[1])) {
+            //       supported = false;
+            //       i = setVersions.length;
+            //     }
+            //   } else if ((">" + actualVer[0]).localeCompare(compareVer[0])) {
+            //     if (actualVer[1].localeCompare(compareVer[1])) {
+
+            //     }
+            //   } else if (">=" +
+
+            //   } else if ("<" + 
+
+            //   } else if ("<=" + 
+
+            //   }
+            // }
+          }
+          if (acceptedCPU) {
+            if (acceptedCPU.includes(components[AGENT_COMP_ID].cpu)) {
+              console.log(`(${components[AGENT_COMP_ID].cpu}) is a supported architecture for the agent to install (${plugin.identifier}).`);
+            } else if (acceptedCPU.includes("!"+components[AGENT_COMP_ID].cpu)) {
+              plugin.error = new Error(`(${components[AGENT_COMP_ID].cpu}) is not a supported architecture for the agent. Skipping (${plugin.identifier})... Supported: ${acceptedCPU}`);
+            }
+            else {
+              console.log(`Agent could not verify (${components[AGENT_COMP_ID].cpu}) as a supported architecture to install (${plugin.identifier}). Proceeding anyway...`);
+            }
+          }
+          if (neededEndpoints && neededEndpoints.length > 0 && components[AGENT_COMP_ID].endpoints) {
+            for (let i = 0; i < neededEndpoints.length; i++) {
+              if (!components[AGENT_COMP_ID].endpoints.includes(neededEndpoints[i])) {
```

Comment:
Actually, Agent's enpoints are urlMasks(e.g. `/datasetMetadata/**`) so you should check matches against URL masks.